frontend/lib/qr_code_scanner/qr_code_scanner_page.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+import 'qr_code_processor.dart';
+import 'qr_code_scanner.dart';
+
+class QrCodeScannerPage extends StatelessWidget {
+  final OnCodeScannedCallback onCodeScanned;
+  final String title;
+
+  QrCodeScannerPage({Key key, @required this.onCodeScanned,
+    @required this.title})
+      : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+          appBar: AppBar(
+          title: Text(title),
+      ),
+      body: QrCodeScanner(
+        onCodeScanned: (code) async => await _onCodeScanned(context, code),
+      )
+    );
+  }
+
+  Future<void> _onCodeScanned(BuildContext context, String code) async {
+    try {
+      if (onCodeScanned != null) {
+        await onCodeScanned(code);
+      }
+    } on QrCodeParseException catch (e, stackTrace) {
+      debugPrintStack(stackTrace: stackTrace, label: e?.toString());
+    }
+
+    Navigator.of(context).maybePop();
+  }
+}
```

Comment:
Is it intended that we leave the qr code scanner on error?

Reply:
Well, it is, but I would also be happy with leaving it open. What do you prefer?

Reply:
For testing it was always useful that it remained open. However, idk if that is useful in practice. Probably not? I mean it's easy to change if we need to

Reply:
Okay. I will just keep it as is for now then